src/App.tsx

```diff
@@ -1,27 +1,95 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
+import { ClockName } from './components/ClockName';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
+const watchNames = [
+  'Omega',
+  'Breguet',
+  'Seiko',
+  'Rolex',
+  'Patek Philippe',
+  'Chopard',
+  'Cartier',
+  'Tissot',
+  'Hublot',
+  'Swiss',
+  'Orient',
+];
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: watchNames[Math.round(Math.random() * 10)],
+  };
 
-  // eslint-disable-next-line
-  console.log(timerId);
+  createRandomName = () => {
+    this.setState({
+      clockName: watchNames[Math.round(Math.random() * 10)],
+    });
+  };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  changeClockVisible = () => {
+    if (this.state.isClockVisible) {
+      this.setState({
+        isClockVisible: false,
+      });
+    } else {
+      this.setState({
+        isClockVisible: true,
+      });
+    }
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        <button
+          type="button"
+          className="Clock__button Clock__button__setName"
+          onClick={this.createRandomName}
+        >
+          Set random name
+        </button>
+        <div className="Clock">
+          <ClockName name={this.state.clockName} />
+          <div className="Clock__face-container">
+            <div
+              className="Clock__face"
+              style={this.state.isClockVisible
+                ? ({ transform: 'translateZ(50px) rotateX(0deg)' })
+                : ({ transform: 'translateZ(-50px) rotateX(90deg' })}
+            >
+              <div>
+                <p className="Clock__face__text" data-cy="time">
+                  {this.state.isClockVisible && (
```

Comment:
better to destructure all variables from state
```suggestion
                  {isClockVisible && (
```